go/mysql/flavor.go

```diff
@@ -75,6 +75,15 @@ type flavor interface {
 	// primaryGTIDSet returns the current GTIDSet of a server.
 	primaryGTIDSet(c *Conn) (GTIDSet, error)
 
+	// purgedGTIDSet returns the purged GTIDSet of a server.
+	purgedGTIDSet(c *Conn) (GTIDSet, error)
+
+	// gtidMode returns the gtid mode of a server.
+	gtidMode(c *Conn) (string, error)
+
+	// serverUUID returns the UUID of a server.
+	serverUUID(c *Conn) (string, error)
+
```

Comment:
technically these are MySQL-specific terms which breaks the "abstraction" of the interface. At this point, it doesn't really matter, since for all practical purposes we can only support MySQL/Percona.